sklearn/metrics/_ranking.py

```diff
@@ -1471,8 +1472,72 @@ def dcg_score(
         ),
         weights=sample_weight,
     )
+from sklearn.inspection import partial_dependence
 
+def h_statistic(dataset, estimator):
```

Comment:
Few thoughts on API:

1. I think needing the estimator puts this into `sklearn.inspection` module.
2. The signature should be something like: `friedman_h(estimator, X, y)` instead of a dataset object. This way it works on normal `X, y` data.
3. It's worth consider `grid_resolution` as a parameter to pass to `partial_dependence` as well.